src/App.tsx

```diff
@@ -42,54 +66,95 @@ export function getReorderedGoods(
 }
 
 // DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    sortType: 0,
+    isReversed: false,
+  };
+
+  reverse = () => {
+    this.setState((state) => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  sortAlphabetically = () => {
+    this.setState({ sortType: 1 });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortType: 2 });
+  };
+
+  reset = () => {
+    this.setState({
+      sortType: 0,
+      isReversed: false,
+    });
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+    const goods = getReorderedGoods(goodsFromServer, { sortType, isReversed });
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            onClick={this.sortAlphabetically}
+            type="button"
+            className={`button is-info ${sortType === 1 ? '' : 'is-light'}`}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            onClick={this.sortByLength}
+            type="button"
+            className={`button is-success ${sortType === 2 ? '' : 'is-light'}`}
+          >
+            Sort by length
+          </button>
+
+          <button
+            onClick={this.reverse}
+            type="button"
+            className={`button is-warning ${isReversed ? '' : ' is-light'}`}
+          >
+            Reverse
+          </button>
+          {sortType === 0 && isReversed === false
+            ? ''
+            : (
```

Comment:
It would be great to create a variable for that condition. Also, you can you logical && instead of the ternary operator as you only decide whether to show the button or not
https://react.dev/learn/conditional-rendering#logical-and-operator-